probe-rs/src/probe/cmsisdap/commands/mod.rs

```diff
@@ -135,6 +139,30 @@ impl CmsisDapDevice {
         }
     }
 
+    /// Determine the correct packet size for this device.
+    ///
+    /// The resulting size is set in the device (either max_packet_size for V2 devices,
+    /// or report_size for V1 devices) and returned.
+    pub(super) fn set_packet_size(&mut self) -> Result<usize, CmsisDapError> {
```

Comment:
I'd call this something else that `set_` as, to me, this suggests that this sets the report size on the debug  probe, which is not the case.
I am not sure what's best. Not sure `read_` or `query_` is better

Reply:
It does set the size we use to talk to the device though (as well as returning it), maybe `get_and_set_packet_size()`?

Reply:
Yah I am aware of the dilemma :P
`get_and_set_packet_size()` sounds better to me imo.

Reply:
Maybe `determine_packet_size`? I feel `get_and_set_packet_size` is not any clearer than just `get_`.

Reply:
I've refactored it slightly so the packet size is determined outside of this method, which now just sets the size on the device struct depending on whether it's a V1 or V2 device. I think this is a bit clearer and now the name `set_packet_size` should make sense too, what do you think?